### ISSUE TITLE
How to represent license choice (disjunctively licensed software)

### ISSUE THREAD

**Issue by KaiHa** [NixOS/nixpkgs#27566]
## Issue description

There exists software (like _libgmp_ [1]) that gives you the choice between multiple licenses. According to the _SPDX_ specification [2] this is expressed through an _OR_ connection. I understand that an _AND_ connection of licenses is already available in _Nix_ through lists. But would it be possible to implement disjunction (choice)? 

If someone points me into the right direction I might give it a try.

[1] https://gmplib.org/
[2] https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60

**Comment by volth** [comment]
the only license which is interpreted by scripts is ```unfree```. All the rest are just a free text, which does not obey any schema.

**Comment by prusnak** [comment]
The answer to the question has been posted. Closing.

**Comment by Bert-Proesmans** [comment]
tldr; assign a list to meta.licenses, because that implies an **or** relationship.

The mentioned library is already packaged in nixpkgs- search for "gmp" and "gmp-with-cxx" to observe single licensed and dual licensed software packaged.

**Comment by MangoIV** [comment]
@prusnak where has the answer been posted?

**Comment by MangoIV** [comment]
Can we reopen this issue? This is a very common thing and it is documented nowhere. I think it should be clarified and documented.

**Comment by MangoIV** [comment]
I do not know who can say something authoritative about this, otherwise I would just document this.
